### PR TITLE
Fixing code checklist

### PR DESCRIPTION
**Pull request by shanioffe**

### CODE REVIEW

File: content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js [hunk 1]
```diff
@@ -16,115 +16,147 @@ $(document).ready(function() {
...
 	    var position = span.position();
 	    var theHeight = $('.popover').height();
 	    $('.rangepop').show();
 			$(".popover").css({
             'position': 'absolute',
             'top': (position.top-(theHeight/2)-5) + 'px',
-            'left': (position.left+50) + 'px'
-      });
+            'left': (position.left+100) + 'px'
+    	});
```
Comment (shanioffe) [inline content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js:160]: If you see a popover covering the code, adjust the pixels here to move it more to the left

File: content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js [hunk 2]
```diff
@@ -16,115 +16,147 @@ $(document).ready(function() {
...
 
 				// if it is, go to next question
-				if(finished) current = advance(current);
+				if (finished) {
+					// highlights correct answer before moving on to next question
+					setTimeout(function(){
+						previousWaitingOnCount += waitingOn.length;
+						current = advance(current);
+					}, 500);
+				}
```
Comment (shanioffe) [inline content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js:145]: If you have a question that only needs you to click one span, the span does not get highlighted before the next question becomes active. You'll need to add this timeout so the user can quickly see that they clicked the right answer.

This goes hand-in-hand with 
`//removes highlighted sections so user is not seeing correct answers of previous question
$(".span-" + name).removeClass("si-code-clicked");`
If the checklist has only multi-span questions, this code can go at the end of `advance()` above the return statement.
If the checklist contains single-span questions, this code goes at the beginning of `advance()` under `var next;`

File: content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js [hunk 3]
```diff
@@ -16,115 +16,147 @@ $(document).ready(function() {
...
 		spans.push($(this));
-		//console.log(spans);
+		
 		$(this).on('click', function() {
 			span = $(this);
 
 			// continue if user is currently supposed to click this span, and hasn't already
-			if( span.hasClass(current+"-"+name) && $.inArray(index,clicked) < 0 ) {
-				clicked.push(index);					// 		note that it's been clicked, programmatically
-				span.addClass("si-code-clicked");		//		note that it's been clicked, graphically
```
Comment (shanioffe) [inline content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js:100]: This was not supposed to get deleted i have to add it back in, ignore this deletion

File: content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js [hunk 4]
```diff
@@ -16,115 +16,147 @@ $(document).ready(function() {
...
-				if(span.hasClass(name+"-var-modify-"+name)|| span.hasClass(name+"-var-array-"+name)) { //needs to be changed for vulnerabilities
-					span.addClass("si-code-vulnerability"); //	some spans get extra graphics to indicate vulnerability
-				}
-
```
Comment (shanioffe) [inline content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js:104]: If you see code highlighted in yellow when the user isn't being asked to identify a code vulnerability, remove this whole statement. Correct answers should be green

File: content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js [hunk 5]
```diff
@@ -16,115 +16,147 @@ $(document).ready(function() {
...
+			waitingOn = [];
+			$(".span-"+name).each(function(index) {
+				if( $(this).hasClass(next+"-"+name) ) {
+					waitingOn.push(index);
 				}
+			});
+		}	
+		//removes highlighted sections so user is not seeing correct answers of previous question
+		$(".span-" + name).removeClass("si-code-clicked");
+		return next;
```
Comment (shanioffe) [inline content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js:110]: This is where this code block should appear if the checklist only has multi-span questions. At the moment this statement is shown in both spots. If the checklist has single-span questions, you would delete this and keep the one that's at the top of this function

File: content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js [hunk 6]
```diff
@@ -16,115 +16,147 @@ $(document).ready(function() {
...
+		if(next === name+"-var-range") { 
+			
+			// If popover is needed, remove progress bar
+			if (
+			next.endsWith("var-occur") || 
+			next.endsWith("var-range")
+			) {  // remove progress bar
+				$("#"+next+"-progress-label").removeClass("progress");
+				$("#"+next+"-progress-label").addClass("progress-hidden");
+			}
```
Comment (shanioffe) [inline content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js:75]: This makes sure the progress bar does not show for popovers

File: content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js [hunk 7]
```diff
@@ -16,115 +16,147 @@ $(document).ready(function() {
...
+		}
+
+		// focus on next question
+		$("#"+next+"-label").addClass("si-checklist-active");
+		// if the question only has 1 span to click, don't show progress bar
+		if(next.endsWith("-array-dec") || next.endsWith("-array-ref") || next.endsWith("-var-array")){ }else{
+			// remove hidden class from progress bar
+			$("#" + next + "-progress-label").removeClass("progress-hidden");
+			// add the progress bar fill
+			$("#" + next + "-progress-label").addClass("progress");
```
Comment (shanioffe) [inline content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js:62]: This makes sure progress bar does not show when there's only 1 span to click

**Comment by pzalep1** [comment]
Testing so far:

**Input Validation**
1. Input Validation-CS0 Pseudocode - Checklist shows progress bar correctly and checks box, but when I select to check answers it says I am incorrect. 
Screenshot 2023-01-29 at 5 31 38 PM

2. Input Validation-CS0 Python -  Progress bar appears and says I finish but when I select to check answers the question won't validate.
Screenshot 2023-01-29 at 5 34 09 PM

When I check the checkAnswers function in AngularJS I noticed that the types for the item.value and item.ans are not the same anymore. I went and checked what is currently in main and both value and answer are boolean values. 
Screenshot 2023-01-29 at 5 35 02 PM

The same issue is present in the rest of the Injections as well.

**Comment by shanioffe** [comment]
Replaced with paigezaleppa/fixing-checklist

**Pull request closed without merging**